pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/IndexCreationContext.java

```diff
@@ -102,7 +102,8 @@ public Builder withColumnMetadata(ColumnMetadata columnMetadata) {
           .withTotalDocs(columnMetadata.getTotalDocs())
           .withDictionary(columnMetadata.hasDictionary())
           .withMinValue(columnMetadata.getMinValue())
-          .withMaxValue(columnMetadata.getMaxValue());
+          .withMaxValue(columnMetadata.getMaxValue())
+          .withMaxNumberOfMultiValueElements(columnMetadata.getMaxNumberOfMultiValues());
```

Comment:
Since this PR is adding support for SV, can we avoid this change ? No harm but would prefer to keep all changes for SV and MV to be localized in their own separate PRs if possible for easy review / tracking etc.

Reply:
Done